sns_ik_lib/src/sns_ik.cpp

```diff
@@ -158,16 +161,39 @@ namespace sns_ik {
         m_types.push_back(SNS_IK::JointType::Prismatic);
       }
     }
-    ROS_ASSERT_MSG(m_types.size()==(unsigned int)m_lower_bounds.data.size(), \
+    ROS_ASSERT_MSG(m_types.size()==(unsigned int)m_lower_bounds.data.size(),
                    "SNS_IK: Could not determine joint limits for all non-continuous joints");
 
     m_jacobianSolver = std::shared_ptr<KDL::ChainJntToJacSolver>(new KDL::ChainJntToJacSolver(m_chain));
-    m_ik_vel_solver = std::shared_ptr<SNSVelocityIK>(new SNSVelocityIK(m_chain.getNrOfJoints(), 0.01)); //TODO make loop rate configurable
+    ROS_ASSERT_MSG(setVelocitySolveType(m_solvetype, 0.01),
+                   "SNS_IK: Failed to create a new SNS velocity and position solver."); //TODO make loop rate configurable
+  }
+
+bool SNS_IK::setVelocitySolveType(VelocitySolveType type, double loop_rate) {
+  // If the requested solve type is different or there is no velocity solver
+  if(m_solvetype != type || !m_ik_vel_solver){
+    switch (type) {
+      case sns_ik::SNS_OptimalScaleMargin:
+        m_ik_vel_solver = std::shared_ptr<OSNS_sm_VelocityIK>(new OSNS_sm_VelocityIK(m_chain.getNrOfJoints(), loop_rate));
+        ROS_INFO("SNS_IK: Set Velocity solver to SNS Optimal Scale Margin solver.");
+        break;
+      case sns_ik::SNS_Optimal:
+        m_ik_vel_solver = std::shared_ptr<OSNSVelocityIK>(new OSNSVelocityIK(m_chain.getNrOfJoints(), loop_rate));
+        ROS_INFO("SNS_IK: Set Velocity solver to SNS Optimal solver.");
+        break;
+      default: // sns_ik::SNS
```

Comment:
I would would change this to "case sns_ik::SNS:" and  have the default throw an error if the solvetype was not set to something in the defined enum.